storage/factory/dbConfigHandler.go

```diff
@@ -0,0 +1,123 @@
+package factory
+
+import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/multiversx/mx-chain-core-go/core"
+	"github.com/multiversx/mx-chain-go/config"
+)
+
+const (
+	dbConfigFileName         = "config.toml"
+	defaultType              = "LvlDBSerial"
+	defaultBatchDelaySeconds = 2
+	defaultMaxBatchSize      = 100
+	defaultMaxOpenFiles      = 10
+)
+
+type dbConfigHandler struct {
+	dbType              string
+	batchDelaySeconds   int
+	maxBatchSize        int
+	maxOpenFiles        int
+	shardIDProviderType string
+	numShards           int32
+}
+
+// NewDBConfigHandler will create a new db config handler instance
+func NewDBConfigHandler(config config.DBConfig) *dbConfigHandler {
+	return &dbConfigHandler{
+		dbType:              config.Type,
+		batchDelaySeconds:   config.BatchDelaySeconds,
+		maxBatchSize:        config.MaxBatchSize,
+		maxOpenFiles:        config.MaxOpenFiles,
+		shardIDProviderType: config.ShardIDProviderType,
+		numShards:           config.NumShards,
+	}
+}
+
+// GetDBConfig will get the db config based on path
+func (dh *dbConfigHandler) GetDBConfig(path string) (*config.DBConfig, error) {
+	dbConfigFromFile := &config.DBConfig{}
+	err := core.LoadTomlFile(dbConfigFromFile, getPersisterConfigFilePath(path))
+	if err == nil {
+		log.Debug("GetDBConfig: loaded db config from toml config file", "path", dbConfigFromFile)
+		return dbConfigFromFile, nil
+	}
+
+	empty := checkIfDirIsEmpty(path)
+	if !empty {
```

Comment:
Is this ok? If not empty load the default config? In which case do you load config from file and when do you use the one from main config file and why?

Reply:
if not empty it will load the default, that means it already running with the current (default) setup, if it is a new epoch, it will be an empty dir so it will load the one from main config file; it will load from toml config file if the node will be started until next epoch change